v3/rfcs/dx/20240620-simplify-subgraph-getting-started-and-federation.md

```diff
@@ -0,0 +1,47 @@
+# DX: simplify subgraphs for getting started and federation workflows
+
+Status: Review required
+
+## subgraph init command
+What this is: as a part of making it easier for people to work through the getting started guide, or to scaffold project files, allow the user to go from scratch to an API with only CLI commands.
+
+init-ing a supergraph or connector will create files for both local and cloud environments. This is about what happens for the subgraphs.
+
+```bash
+# creates a subgraph folder and subgraph.yaml file
+ddn subgraph init app
+
+# add subgraph.yaml to the supergraph.yaml file and create an env file for the env
+ddn supergraph add subgraph --supergraph supergraph.local.yaml --subgraph app/subgraph.yaml
+touch app/.env.app.local
+
+# create the supergraph build
+ddn supergraph build --supergraph supergraph.yaml --env-file app/.env.app.local
```

Comment:
In supergraph build, there could be multiple subgraphs and each can have its own env file. With this command, we cannot determine, which env file belongs to which `subgraph.yaml`. We might need to do something like the prefixing we did for the `-e` flag like `--env-file <subgraph-Name>=<path-to-env-file>`. Looks ugly though

Reply:
You're right. Can we do `--env-file <subgraph>:<env file>`.

Can do `--env <subgraph>:<key>=<value>` for consistency.